pkg/cli/adbtunnel.go

```diff
@@ -223,48 +236,256 @@ func (f *ADBForwarder) Send(data []byte) error {
 	return nil
 }
 
-func (f *ADBForwarder) acceptLoop(listener net.Listener) {
-	defer listener.Close()
+type StatusMsg struct {
+	ServiceURL string `json:"service_url"`
+	Zone       string `json:"zone"`
+	Host       string `json:"host"`
+	Device     string `json:"device"`
+	Port       int    `json:"port"`
+	Status     string `json:"status"`
+}
+
+func (f *ADBForwarder) StatusJSON() []byte {
+	f.statusMtx.Lock()
+	status := f.status
+	f.statusMtx.Unlock()
+
+	msg := StatusMsg{
+		ServiceURL: f.serviceURL,
+		Zone:       f.zone,
+		Host:       f.host,
+		Device:     f.device,
+		Port:       f.port,
+		Status:     status,
+	}
+	ret, err := json.Marshal(&msg)
+	if err != nil {
+		panic("Couldn't marshal status map")
```

Comment:
nit: add the error text to this message.

Reply:
Done